core/src/model/api.rs

```diff
@@ -160,6 +163,7 @@ pub enum NewAccountError {
     InvalidAuth,
     ExpiredAuth,
     UsernameTaken,
+    InvalidUsername,
```

Comment:
I think this is actually the only place we need to add this error. If someone passes an invalid username to CreateFile, that username won't be found because we wouldn't have allowed it to be created, and we'd return UserNotFound

Reply:
ah okay